perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,246 @@
+      table = client.get_table(table_ref)
+      row_count = table.num_rows
+      config_id = row_count + 1
+      rows_to_insert = [(config_id, gcsfuse_flags, branch, end_date)]
+      results = client.insert_rows(table, rows_to_insert)
+
+    return int(config_id)
+
+  def _setup_bigquery(self):
+
+    """Creates the tables to store the metrics data if they don't exist in the dataset
+    """
+    # Query for creating fio_metrics table
+    query_create_table_fio_metrics = """
+          CREATE TABLE IF NOT EXISTS {}.{}.{}(
+            configuration_id INT64, 
+            start_time_build TIMESTAMP,
+            test_type STRING, 
+            num_threads INT64, 
+            file_size_kb INT64, 
+            block_size_kb INT64,
+            start_time INT64, 
+            end_time INT64, 
+            iops FLOAT64, 
+            bandwidth_bytes_per_sec INT64, 
+            IO_bytes INT64, 
+            min_latency FLOAT64, 
+            max_latency FLOAT64, 
+            mean_latency FLOAT64, 
+            percentile_latency_20 FLOAT64, 
+            percentile_latency_50 FLOAT64, 
+            percentile_latency_90 FLOAT64, 
+            percentile_latency_95 FLOAT64, 
+            FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+          ) OPTIONS (description = 'Table for storing FIO metrics extracted from periodic performance load testing');
+      """.format(PROJECT_ID, DATASET_ID, FIO_TABLE_ID, DATASET_ID, CONFIGURATION_TABLE_ID)
+
+    # Query for creating vm_metrics table
+    query_create_table_vm_metrics = """
+          CREATE TABLE IF NOT EXISTS {}.{}.{}(
+            configuration_id INT64, 
+            start_time_build TIMESTAMP,
+            end_time INT64, 
+            cpu_utilization_peak_percentage FLOAT64, 
+            cpu_utilization_mean_percentage FLOAT64, 
+            received_bytes_peak_bytes_per_sec FLOAT64, 
+            received_bytes_mean_bytes_per_sec FLOAT64, 
+            read_bytes_count INT64,
+            ops_error_count INT64, 
+            ops_mean_latency_sec FLOAT64, 
+            sent_bytes_per_sec FLOAT64, 
+            memory_utilization_ram FLOAT64,
+            memory_utilization_disk_tempdir FLOAT64,
+            iops FLOAT64, 
+            ops_count_list_object INT64, 
+            ops_count_create_object INT64, 
+            ops_count_stat_object INT64, 
+            ops_count_new_reader INT64, 
+            FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+          ) OPTIONS (description = 'Table for storing VM metrics extracted from periodic performance load testing');
+      """.format(PROJECT_ID, DATASET_ID, VM_TABLE_ID, DATASET_ID, CONFIGURATION_TABLE_ID)
+
+    # Query for creating ls_metrics table
+    query_create_table_ls_metrics = """
+          CREATE TABLE IF NOT EXISTS {}.{}.{}(
+            configuration_id INT64,
+            start_time_build TIMESTAMP,
+            test_type STRING, 
```

Comment:
what does test_type mean here ?

Reply:
test_type is 'gcs bucket' or 'persistent_disk'

Reply:
got it, let's call it "mount_type" and the values can be "gcs_bucket" or "persistent_disk"